flocker/node/_deploy.py

```diff
@@ -766,58 +809,14 @@ def calculate_necessary_state_changes(self, local_state,
                 changes = [
                     StopApplication(application=inspect_current),
                     StartApplication(application=inspect_desired,
-                                     hostname=self.hostname)
+                                     node_state=current_node_state),
                 ]
                 sequence = Sequentially(changes=changes)
                 if sequence not in restart_containers:
                     restart_containers.append(sequence)
 
-        # Find any dataset that are moving to or from this node - or
-        # that are being newly created by this new configuration.
-        dataset_changes = find_dataset_changes(
-            self.hostname, current_cluster_state, desired_configuration)
-
-        if dataset_changes.resizing:
-            phases.append(InParallel(changes=[
-                ResizeDataset(dataset=dataset)
-                for dataset in dataset_changes.resizing]))
-
-        # Do an initial push of all volumes that are going to move, so
-        # that the final push which happens during handoff is a quick
-        # incremental push. This should significantly reduces the
-        # application downtime caused by the time it takes to copy
-        # data.
-        if dataset_changes.going:
-            phases.append(InParallel(changes=[
-                PushDataset(dataset=handoff.dataset,
-                            hostname=handoff.hostname)
-                for handoff in dataset_changes.going]))
-
         if stop_containers:
             phases.append(InParallel(changes=stop_containers))
-        if dataset_changes.going:
-            phases.append(InParallel(changes=[
-                HandoffDataset(dataset=handoff.dataset,
-                               hostname=handoff.hostname)
-                for handoff in dataset_changes.going]))
-        # any datasets coming to this node should also be
-        # resized to the appropriate quota max size once they
-        # have been received
-        if dataset_changes.coming:
-            phases.append(InParallel(changes=[
-                WaitForDataset(dataset=dataset)
```

Comment:
Just for your consideration - deletion of this part of the convergence logic, along with all of the support code, might have made for a nice separate PR.  It seems independent and it would have knocked a couple hundred lines off this diff (though, admittedly, pretty much all `-`s so the effect may not have been particularly great).